### PR TITLE
ODE and Tau Leaping C++ solvers

### PR DESCRIPTION
**Pull request by Fin109**
Adds TauLeaping C++ and ODE C++ solvers to GillesPy2

- TODO: Fix unit tests
- Ensure absolute path is working

### CODE REVIEW

File: gillespy2/solvers/cpp/c_base/tau_leaping_cpp_solver/tau_leaper.cpp [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+        min = *min_element(critical_taus.begin(), critical_taus.end(),[](const auto& lhs, const auto& rhs){ return lhs.second < rhs.second;});
+        critical_tau = min.second;
+    }
+
+    if (tau_args.g_i_lambdas.size()>0){
+        for (auto const& x : tau_args.g_i_lambdas)
+           {
+               tau_args.g_i[x.first] = tau_args.g_i_lambdas[x.first](tau_args.g_i[x.first]);
+               tau_args.epsilon_i[x.first] = tau_tol / tau_args.g_i[x.first];
+               tau_args.g_i_lambdas.erase(x.first);//MAYBE SHOULDN'T ERASE, may break loop/may be understanding this part wrong
```
Comment (seanebum) [inline gillespy2/solvers/cpp/c_base/tau_leaping_cpp_solver/tau_leaper.cpp:138]: Regarding code comment - Is this safe / has this been tested?

Comment (Fin109) [reply gillespy2/solvers/cpp/c_base/tau_leaping_cpp_solver/tau_leaper.cpp:138]: Yes, this had to do with something else I was testing, I seem to have left the comment in -- I'll delete it

File: gillespy2/solvers/cpp/tau_leaping_c_solver.py [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+
+GILLESPY_PATH = os.path.dirname(os.path.abspath(__file__))
+GILLESPY_CPP_TAU_DIR = os.path.join(GILLESPY_PATH, 'c_base/tau_leaping_cpp_solver')
+MAKE_FILE = os.path.dirname(os.path.abspath(__file__)) + '/c_base/tau_leaping_cpp_solver/makefile'
+CBASE_DIR = os.path.join(GILLESPY_PATH, 'c_base/')
+
+
+class TauLeapingCSolver(GillesPySolver):
+    name = "TauLeapingCSolver"
+    """TODO"""
```
Comment (seanebum) [inline gillespy2/solvers/cpp/tau_leaping_c_solver.py:20]: Comment cleanup

File: gillespy2/solvers/cpp/ode_c_solver.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+GILLESPY_PATH = os.path.dirname(os.path.abspath(__file__))
+GILLESPY_C_ODE_DIR = os.path.join(GILLESPY_PATH, 'c_base/ode_cpp_solver')
+MAKE_FILE = os.path.dirname(os.path.abspath(__file__)) + '/c_base/ode_cpp_solver/makefile'
+SUNDIALS_DIR = os.path.join(GILLESPY_PATH, 'c_base/Sundials')
+CBASE_DIR = os.path.join(GILLESPY_PATH, 'c_base/')
+
+class ODECSolver(GillesPySolver):
+    name = "ODECSolver"
+    """TODO"""
```
Comment (seanebum) [inline gillespy2/solvers/cpp/ode_c_solver.py:19]: comment cleanup

**Review by seanebum — CHANGES_REQUESTED** [review]
left a couple of single line comments related to comments in the code - but otherwise seems good to go to me, works great for me!

**Review by BryanRumsey — APPROVED** [review]

**Review by seanebum — APPROVED** [review]

**Pull request merged**